network_snapshot/network_snapshot.py

```diff
@@ -0,0 +1,441 @@
+#!/usr/bin/env python3
+
+import argparse
+import csv
+import gzip
+import json
+import logging
+import logging.handlers
+import os
+import re
+import requests
+
+from collections import defaultdict
+from decimal import Decimal
+from glob import glob
+from os import path
+from time import sleep
+from multiprocessing.pool import ThreadPool
+
+
+base = path.dirname(path.realpath(__file__))
+data = path.abspath(path.join(base, 'data'))
+
+block_log = path.join(data, 'block.log')
+rpc_log = path.join(data, 'account_rpc.log')
+validator_log = path.join(data, 'validator.log')
+
+headers = {'Content-Type': 'application/json'}
+
+# Autonode/util.py
+class _GZipRotator:
+    """A simple zip rotator for logging"""
+
+    def __call__(self, source, dest):
+        os.rename(source, dest)
+        f_in = open(dest, 'rb')
+        f_out = gzip.open("%s.gz" % dest, 'wb')
+        f_out.writelines(f_in)
+        f_out.close()
+        f_in.close()
+        os.remove(dest)
+
+def get_simple_rotating_log_handler(log_file_path):
+    """
+    A simple log handler with no level support.
+    Used purely for the output rotation.
+    """
+    log_formatter = logging.Formatter('%(message)s')
+    handler = logging.handlers.TimedRotatingFileHandler(log_file_path, when = 'h', interval = 2,
+                                                        backupCount = None, encoding = 'utf8')
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(log_formatter)
+    handler.rotator = _GZipRotator()
+    return handler
+
+def get_node_metadata(endpoint, retries):
```

Comment:
consider using the pyhmy package DM created.

Reply:
The pyhmy lib uses the CLI to send requests, which is slow